backend/btrixcloud/pages.py

```diff
@@ -591,8 +603,20 @@ async def list_pages(
 
                 query[f"qa.{qa_run_id}.{qa_filter_by}"] = range_filter
 
-        aggregate = [{"$match": query}]
+        aggregate: List[Dict[str, Union[int, object]]] = [{"$match": query}]
 
+        # If total, compute here first
+        if include_total:
+            total = await self.pages.count_documents(query)
+        else:
+            total = 0
+
+        # Extra QA Set
+        if qa_run_id:
+            aggregate.extend([{"$set": {"qa": f"$qa.{qa_run_id}"}}])
+            # aggregate.extend([{"$project": {"qa": f"$qa.{qa_run_id}"}}])
+
+        # Sorting
```

Comment:
Sorting options from `list_replay_query_pages` that aren't here yet but we might want to add:
- `url`
- `crawl_id`

Reply:
`url` is in the sort_fields below, could add `crawl_id` also..